File: Backend/src/main/java/com/back/domain/order/dto/OrderResponse.java
package com.back.domain.order.dto;

import com.back.domain.order.entity.Order;
import lombok.Getter;

import java.time.LocalDate;
import java.util.List;

@Getter
public class OrderResponse {
    private String address;
    private String zipCode;
    private String email;
    private int totalQuantity;
    private int totalPrice;
    private LocalDate deliveryDate;
    private String deliveryStatus;
    private List<OrderProductDto> products;

    public OrderResponse(Order order) {
        this.address = order.getAddress();
        this.zipCode = order.getZipCode();
        this.email = order.getEmail();
        this.totalQuantity = order.getTotalQuantity();
        this.totalPrice = order.getTotalPrice();
        this.deliveryDate = order.getDeliveryDate();
        this.deliveryStatus = order.getDeliveryStatus();
        this.products = order.getOrderProducts().stream()
                .map(OrderProductDto::new)
                .toList();
    }
}

File: Frontend/front-app/src/type/orderResponse.ts
import { OrderProductDto } from "./orderProduct"; // 가정

export type OrderResponseDto = {
    totalQuantity: number;
    totalPrice: number;    
    deliveryDate: string;     
    deliveryStatus: string; 
    address: string;
    zipCode : string;
    email : string;
    // OrderProductDto 리스트 포함
    products: OrderProductDto[]; 
};